### PR TITLE
fix: applicant - invalidate cache to prevent wrong application mode (HL-1183)

### PR DESCRIPTION
**Pull request by sirtawast**
## Description :sparkles:

Invalidate applications cache on frontpage to prevent wrong state being loaded.

### CODE REVIEW

**Comment by terovirtanen** [comment]
**API branch is deployed to platta: https://helsinkilisa-pr2901.api.dev.hel.ninja/healthz** :rocket::rocket::rocket:

File: frontend/benefit/applicant/src/components/applications/applicationList/listItem/ListItem.tsx [hunk 1]
```diff
@@ -1,3 +1,5 @@
+import 'react-loading-skeleton/dist/skeleton.css';
```
Comment (EmiliaMakelaVincit) [inline frontend/benefit/applicant/src/components/applications/applicationList/listItem/ListItem.tsx:1]: Placing this import here is a bit curious. My first instinct would be to put it where all other library CSS imports are, to `_app.tsx`; is there a reason you'd put it specifically here instead, other than that the skeleton is relevant to this component?

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://helsinkilisa-ui-pr2901.dev.hel.ninja :laughing::tada::tada::tada:

File: frontend/benefit/applicant/src/pages/index.tsx [hunk 2]
```diff
@@ -31,6 +34,14 @@ const ApplicantIndex: NextPage = () => {
     };
   }, [setIsNavigationVisible]);
 
+  /**
+   * Fix a cache issue where single application's button opens up a read-only
+   * application if status changes to "additional_information_needed"
+   */
+  const effectInvalidateApplicationsCache = () =>
+    void queryClient.invalidateQueries('applications');
+  useEffect(effectInvalidateApplicationsCache, [queryClient]);
```
Comment (EmiliaMakelaVincit) [inline frontend/benefit/applicant/src/pages/index.tsx:43]: The exact conditions that make the query client change and thus trigger this are a bit unclear to me, but if this works, sure.

**Comment by terovirtanen** [comment]
**HANDLER branch is deployed to platta: https://helsinkilisa-ui-handler-pr2901.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://helsinkilisa-ui-handler-pr2901.dev.hel.ninja :laughing::tada::tada::tada:

File: frontend/benefit/applicant/src/components/applications/applicationList/listItem/ListItem.tsx (hunk 1 above)
Comment (sirtawast) [reply frontend/benefit/applicant/src/components/applications/applicationList/listItem/ListItem.tsx:1]: Yes, maybe change that there. I was thinking NextJS is smart enough to do some sort of splitting and lazy load of assets but then again 99% of cases would download that CSS anyway 🤷

File: frontend/benefit/applicant/src/pages/index.tsx (hunk 2 above)
Comment (sirtawast) [reply frontend/benefit/applicant/src/pages/index.tsx:43]: For some reason the navigation results to react-query fetching data from cache even though all requests should be considered as non-stale:

image

I consider it safe to just clear any leftover react-query cache here as it's never meant to work with cache.

https://github.com/TanStack/query/discussions/4666

**Comment by terovirtanen** [comment]
**API branch is deployed to platta: https://helsinkilisa-pr2901.api.dev.hel.ninja/healthz** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**APPLICANT branch is deployed to platta: https://helsinkilisa-ui-pr2901.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://helsinkilisa-ui-pr2901.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**HANDLER branch is deployed to platta: https://helsinkilisa-ui-handler-pr2901.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://helsinkilisa-ui-handler-pr2901.dev.hel.ninja :laughing::tada::tada::tada:

**Review by EmiliaMakelaVincit — APPROVED** [review]

**Pull request merged**